lib/auth/auth_with_roles.go

```diff
@@ -3590,26 +3591,39 @@ func (a *ServerWithRoles) GetDatabaseServers(ctx context.Context, namespace stri
 
 // UpsertDatabaseServer creates or updates a new database proxy server.
 func (a *ServerWithRoles) UpsertDatabaseServer(ctx context.Context, server types.DatabaseServer) (*types.KeepAlive, error) {
+	log.Infof("Add or update database server: %v", server.GetName())
 	if err := a.action(server.GetNamespace(), types.KindDatabaseServer, types.VerbCreate, types.VerbUpdate); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return a.authServer.UpsertDatabaseServer(ctx, server)
+	keepAlive, err := a.authServer.UpsertDatabaseServer(ctx, server)
+	if err == nil {
+		publishAppChanges(a.authServer, publisher.Database)
+	}
+	return keepAlive, err
 }
 
 // DeleteDatabaseServer removes the specified database proxy server.
 func (a *ServerWithRoles) DeleteDatabaseServer(ctx context.Context, namespace, hostID, name string) error {
 	if err := a.action(namespace, types.KindDatabaseServer, types.VerbDelete); err != nil {
 		return trace.Wrap(err)
 	}
-	return a.authServer.DeleteDatabaseServer(ctx, namespace, hostID, name)
+	err := a.authServer.DeleteDatabaseServer(ctx, namespace, hostID, name)
+	if err == nil {
```

Comment:
do you want to log a message in case of an error? same for all the other cases as well in this review.

Reply:
Wanted to figure out which api is invoked and then i will add logs for all later.